org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/CompilationUnitScope.java

```diff
@@ -226,11 +228,35 @@ void checkAndSetImports() {
 		return;
 	}
 
+	int numberOfImports = this.referenceContext.imports.length;
+	ImportBinding[] resolvedNonStaticImports = getResolvedImports(importReference -> !importReference.isStatic());
+	if (resolvedNonStaticImports.length == numberOfImports) {
+		// No static imports, we don't need to do more work here.
+		this.imports = resolvedNonStaticImports;
+	} else {
+		/*
+		 * We have static imports, those need to know the "normal" imports
+		 * in order to be resolved correctly. So temporarily set the imports
+		 * to the non-static ones, and resolve the static ones.
+		 * See: https://github.com/eclipse-jdt/eclipse.jdt.core/issues/269
+		 */
+		this.imports = resolvedNonStaticImports;
+		ImportBinding[] resolvedStaticImports = getResolvedImports(importReference -> !importReference.isStatic());
```

Comment:
is this a typo: `!importReference.isStatic()` ?

Reply:
> is this a typo: `!importReference.isStatic()` ?

Yes.

Reply:
Could you push the fix, I see lot of tests failing. Or will they fail anyway, because of the patch?